DIS-313: Filter Checkouts

AlexanderBlanchardAC opened the pull request:
https://aspen-discovery.atlassian.net/browse/DIS-313

AlexanderBlanchardAC committed:
DIS-313: feat: Add filtering of checkouts to db

-Add a column to the library table in the db to allow users to filter
checkouts by linked user

AlexanderBlanchardAC committed:
DIS-313: feat: Add checkout filtering to object

-Add control over allowing users to filter checkouts by linked accounts
to the library object

AlexanderBlanchardAC committed:
DIS-313: feat: Add filter to check out page

AlexanderBlanchardAC committed:
DIS-313: feat: Filter by linked users in checkouts

AlexanderBlanchardAC committed:
DIS-313: feat: export selected checkouts in db

AlexanderBlanchardAC committed:
DIS-313: feat: Add export selected checkouts to obj

AlexanderBlanchardAC committed:
DIS-313: feat: Add export selected checkouts option

-Add to interface
-Add js function

AlexanderBlanchardAC committed:
DIS-313: feat: Export selected checkouts

AlexanderBlanchardAC committed:
DIS-313: feat: Update counts when filtering checkouts

AlexanderBlanchardAC committed:
DIS-313: refactor: create linked user template

AlexanderBlanchardAC committed:
DIS-313: Refactor: Seperate holds and checkouts filtering

AlexanderBlanchardAC committed:
DIS-313: refactor: update correct db

AlexanderBlanchardAC committed:
DIS-313: chore: Update change log

Jacobomara901 commented:
Hiya bud, mostly looks great and works as expected. I noticed a couple of things that I'll list below!

- 3 commits missing conventional commits (Not sure if we're still enforcing this but just thought I'd point it out)
- DB update still has PTFS (Andrew will be sad)
- If you have multiple version of the same record checked out, selecting one will return all of them. Technically these are seperate checkouts so not sure if we want to be respecting these. The csv export doesn't distinguish between them at all (other that the user they're checked out to)

AlexanderBlanchardAC committed:
refactor: Check record belongs to patron

When filtering by patron, if you select one of their records to export,
ensure that only this record and not any matching records for other
users.

AlexanderBlanchardAC committed:
DIS-313: refactor: Update comapny name in db file

AlexanderBlanchardAC commented:
All issues identified resolved
